src/script/descriptor.cpp

```diff
@@ -625,6 +625,80 @@ std::unique_ptr<Descriptor> ParseScript(Span<const char>& sp, ParseScriptContext
     return nullptr;
 }
 
+std::unique_ptr<PubkeyProvider> InferPubkey(const CPubKey& pubkey, ParseScriptContext, const SigningProvider& provider)
```

Comment:
It can be used for extra sanity checking (like rejecting malformed constructions with an uncompressed pubkey inside a witness) early, instead of having them fail when expanding the inferred descriptor.

Reply:
Could add that note to the code otherwise it can be removed in the future?